ui/src/dashboards/components/TemplateVariableRow.js

```diff
@@ -0,0 +1,472 @@
+import React, {PropTypes, Component} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import OnClickOutside from 'react-onclickoutside'
+
+import Dropdown from 'shared/components/Dropdown'
+import DeleteConfirmButtons from 'shared/components/DeleteConfirmButtons'
+import TemplateQueryBuilder
+  from 'src/dashboards/components/TemplateQueryBuilder'
+
+import {
+  runTemplateVariableQuery as runTemplateVariableQueryAJAX,
+} from 'src/dashboards/apis'
+
+import parsers from 'shared/parsing'
+
+import {TEMPLATE_TYPES} from 'src/dashboards/constants'
+import generateTemplateVariableQuery
+  from 'src/dashboards/utils/templateVariableQueryGenerator'
+
+import {publishAutoDismissingNotification} from 'shared/dispatchers'
+
+const RowValues = ({
```

Comment:
It's hard to tell what's going on in this file. We should extract these components into their own files, maybe grouped in a directory.

```
src/template_variables/components/RowValues.js
src/template_variables/components/RowButtons.js
src/template_variables/components/RowWrapper.js
src/template_variables/components/TemplateVariableRow.js
src/template_variables/components/TableInput.js
```